text/0000-PXE-Booting-UEFI.md

```diff
@@ -0,0 +1,228 @@
+- Feature Name: UEFI support for PXE booting
+- Start Date: 2016-05-04
+- RFC PR: (leave this empty)
+- Issue: [Redmine #431](http://projects.theforeman.org/issues/431)
+
+# Summary
+[summary]: #summary
+
+The goal of this feature is to implement PXE booting of UEFI systems. In the
+default configuration on at least RHEL6, RHEL7 and Debian stable platforms,
+both BIOS and EFI systems should be able to PXE boot including Discovery Image.
+Foreman bootdisk EFI support is out of scope of this task.
+
+# Owners
+
+Lukáš Zapletal (Foreman) - `lzap_at_redhat_dot_com`
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Foreman supports booting of x86 BIOS systems via PXELinux and
+several other platforms via Grub1, ZTP and POAP. These are represented as PXE
+"variants": syslinux, pxegrub, ztp and poap which are hardcoded in operating
+system model classes. However, EFI is not currently supported, and should be.
+
+# Detailed design
+[design]: #detailed-design
+
+Currently, Foreman supports booting of x86 BIOS systems via PXELinux and
+several other platforms via Grub1, ZTP and POAP. These are represented as PXE
+"variants": syslinux, pxegrub, ztp and poap which are hardcoded in operating
+system model classes.
+
+Initial testing was done to get EFI working on PXELinux and it looks like
+stable version of Syslinux (= PXELinux), which is currently 6.02, does not
+appear to support PXE booting correctly (at least not with TianoCore EFI ROM in
+QEMU/KVM). The only version which was found working was 6.03-pre20.
+
+Additionally, there is no PXELinux version with EFI support in RHEL6 and RHEL7
+and both systems ship with Grub (versions 1 and 2 respectively). Grub is more
+mature when it comes to EFI support and since it can boot via PXE and Foreman
+already supports Grub1, it is more feasible to get EFI working via officially
+supported Grub. To have complete support matrix, PXELinux will be part of this
+implementation, but hardware support (and thus testing) will be limited.
+
+The first idea was to simply deploy both BIOS and UEFI configurations during
```

Comment:
I find this section confusion, as it looks like this is part of the solution while it actually not. imho this could be an appendix / optional or something similar but later down the document.